Document how to access rootless containers-storage:

I am currently trying to read images from the local podman container storage via the following code:
```go
package main

import (
	"fmt"

	"github.com/containers/image/v5/transports/alltransports"
)

func main() {
	imgRef, err := alltransports.ParseImageName("containers-storage:registry.opensuse.org/opensuse/busybox:latest")
	if err != nil {
		panic(err)
	}
	fmt.Println(imgRef)
}
```
which fails with:
```
panic: chown /home/dan/.local/share/containers/storage/overlay/l: operation not permitted
```

Running this through strace reveals that the library tries to `chown` the directory to `root:root`, which fails when run as a normal user
```
fchownat(AT_FDCWD, "/home/dan/.local/share/containers/storage/overlay/l", 0, 0, 0) = -1 EPERM (Operation not permitted)
```

Now, given that skopeo can inspect that reference (`skopeo inspect containers-storage:registry.opensuse.org/opensuse/busybox`) leads me to the assumption that I am doing something wrong. Unfortunately, I don't really see what.

--- a/doc.go
+++ b/doc.go
@@ -1,32 +1,71 @@
-// Package image provides libraries and commands to interact with containers images.
+// The package image provides libraries and commands to interact with container images.
 //
 //  package main
 //
 //  import (
-//   	"context"
-//  	"fmt"
+//	"context"
+//	"fmt"
 //
-//  	"github.com/containers/image/v5/docker"
+//	"github.com/containers/image/v5/docker"
 //  )
 //
 //  func main() {
-//  	ref, err := docker.ParseReference("//fedora")
-//  	if err != nil {
-//  		panic(err)
-//  	}
-//  	ctx := context.Background()
-//  	img, err := ref.NewImage(ctx, nil)
-//  	if err != nil {
-//  		panic(err)
-//  	}
-//  	defer img.Close()
-//  	b, _, err := img.Manifest(ctx)
-//  	if err != nil {
-//  		panic(err)
-//  	}
-//  	fmt.Printf("%s", string(b))
+//	ref, err := docker.ParseReference("//fedora")
+//	if err != nil {
+//		panic(err)
+//	}
+//	ctx := context.Background()
+//	img, err := ref.NewImage(ctx, nil)
+//	if err != nil {
+//		panic(err)
+//	}
+//	defer img.Close()
+//	b, _, err := img.Manifest(ctx)
+//	if err != nil {
+//		panic(err)
+//	}
+//	fmt.Printf("%s", string(b))
 //  }
 //
+//
+// ## Notes on running in rootless mode
+//
+// If your application needs to access a containers/storage store in rootless
+// mode, then the following additional steps have to be performed at start-up of
+// your application:
+//
+//  package main
+//
+//  import (
+//  	"github.com/containers/storage/pkg/reexec"
+//	"github.com/syndtr/gocapability/capability"
+//      "github.com/containers/storage/pkg/unshare"
+//  )
+//
+//  var neededCapabilities = []capability.Cap{
+//          capability.CAP_CHOWN,
+//          capability.CAP_DAC_OVERRIDE,
+//          capability.CAP_FOWNER,
+//          capability.CAP_FSETID,
+//          capability.CAP_MKNOD,
+//          capability.CAP_SETFCAP,
+//  }
+//
+//  func main() {
+//          	reexec.Init()
+//
+//        	capabilities, err := capability.NewPid(0)
+//		if err != nil {
+//			panic(err)
+// 		}
+//		for _, cap := range neededCapabilities {
+//			if !capabilities.Get(capability.EFFECTIVE, cap) {
+//				// We miss a capability we need, create a user namespaces
+//				unshare.MaybeReexecUsingUserNamespace(true)
+//			}
+//		}
+//  		// rest of your code follows here
+//  }
 //
 // TODO(runcom)
 package image

